deepchem/models/torch_models/dtnn.py

```diff
@@ -0,0 +1,280 @@
+import numpy as np
+from typing import List
+
+import torch.nn as nn
+import torch
+
+from deepchem.models.losses import L2Loss
+from deepchem.models.torch_models import layers
+from deepchem.models.torch_models import TorchModel
+
+
+class DTNN(nn.Module):
+    """Deep Tensor Neural Networks
+
+    In this class, we establish a sequential model for the Deep Tensor Neural Network (DTNN) [1]_.
+
+    References
+    ----------
+    .. [1] Schütt, Kristof T., et al. "Quantum-chemical insights from deep
+        tensor neural networks." Nature communications 8.1 (2017): 1-8.
+
+    """
+
+    def __init__(
+        self,
+        n_tasks: int,
+        n_embedding: int = 30,
+        n_hidden: int = 100,
+        n_distance: int = 100,
+        distance_min: float = -1,
+        distance_max: float = 18,
+        output_activation: bool = True,
+        mode: str = "regression",
+        dropout: float = 0.0,
+    ):
+        """
+        Parameters
+        ----------
+        n_tasks: int
+            Number of tasks
+        n_embedding: int (default 30)
+            Number of features per atom.
+        n_hidden: int (default 100)
+            Number of features for each molecule after DTNNStep
+        n_distance: int (default 100)
+            granularity of distance matrix
+            step size will be (distance_max-distance_min)/n_distance
+        distance_min: float (default -1)
+            minimum distance of atom pairs (in Angstrom)
+        distance_max: float (default 18)
+            maximum distance of atom pairs (in Angstrom)
+        output_activation: bool (default True)
+            determines whether an activation function should be apply  to its output.
+        mode: str (default "regression")
+            Only "regression" is currently supported.
+        dropout: float (default 0.0)
+            the dropout probablity to use.
+
+        """
+        super(DTNN, self).__init__()
+        if mode not in ['regression']:
+            raise ValueError("Only 'regression' mode is currently supported")
+        self.n_tasks = n_tasks
+        self.n_embedding = n_embedding
+        self.n_hidden = n_hidden
+        self.n_distance = n_distance
+        self.distance_min = distance_min
+        self.distance_max = distance_max
+        self.step_size = (distance_max - distance_min) / n_distance
+        self.steps = np.array(
+            [distance_min + i * self.step_size for i in range(n_distance)])
+        self.steps = np.expand_dims(self.steps, 0)
+        self.output_activation = output_activation
+        self.mode = mode
+        self.dropout = dropout
+        self.use_dropout = False
+
+        if self.dropout > 0.0:
```

Comment:
Also, this was already validated in another function, right?
So remove this too